packages/react-aria-components/stories/Table.stories.tsx

```diff
@@ -393,6 +407,265 @@ const MyCheckbox = ({children, ...props}: CheckboxProps) => {
+      <MyTableBody rows={rows} isLoadingMore={args.isLoadingMore}>
+        {(item) => (
+          <Row columns={columns}>
+            {(column) => {
+              return <Cell>{item[column.id]}</Cell>;
+            }}
+          </Row>
+        )}
+      </MyTableBody>
+    </Table>
+  );
+};
+
+export const TableLoadingBodyWrapperStory = {
+  render: TableLoadingBodyWrapper,
+  args: {
+    isLoadingMore: false
+  },
+  name: 'Table loading, table body wrapper with collection'
+};
+
+function MyRow(props) {
+  return (
+    <>
+      {/* Note that all the props are propagated from MyRow to Row, ensuring the id propagates */}
+      <Row {...props} />
+      {props.isLoadingMore && <MyTableLoader />}
+    </>
+  );
+}
+
+const TableLoadingRowRenderWrapper = (args: {isLoadingMore: boolean}) => {
+  return (
+    <Table aria-label="Files">
+      <TableHeader columns={columns}>
+        {(column) => (
+          <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+        )}
+      </TableHeader>
+      <TableBody items={rows} dependencies={[args.isLoadingMore]}>
+        {(item) => (
+          <MyRow columns={columns} isLoadingMore={item.id === 4 && args.isLoadingMore}>
+            {(column) => {
+              return <Cell>{item[column.id]}</Cell>;
+            }}
+          </MyRow>
+        )}
+      </TableBody>
+    </Table>
+  );
+};
+
+export const TableLoadingRowRenderWrapperStory = {
+  render: TableLoadingRowRenderWrapper,
+  args: {
+    isLoadingMore: false
+  },
+  name: 'Table loading, row renderer wrapper and dep array'
+};
+
+
+function renderEmptyLoader({isLoading, tableWidth = 400}) {
+  let contents = isLoading ? 'Loading spinner' : 'No results found';
+  return <div style={{height: 'inherit', position: 'sticky', top: 0, left: 0, width: tableWidth}}>{contents}</div>;
+}
+
+const RenderEmptyState = (args: {isLoading: boolean}) => {
+  let {isLoading} = args;
+  return (
+    <Table aria-label="Files" selectionMode="multiple">
+      <TableHeader columns={columns}>
+        <Column><MyCheckbox slot="selection" /></Column>
+        <Collection items={columns}>
+          {(column) => (
+
+            <Column isRowHeader={column.isRowHeader}>{column.name}</Column>
+          )}
+        </Collection>
+      </TableHeader>
+      <TableBody renderEmptyState={() => renderEmptyLoader({isLoading})}>
+        <Collection items={[]}>
+          {(item) => (
+            <Row columns={columns}>
+              {(column) => {
+                return <Cell>{item[column.id]}</Cell>;
+              }}
+            </Row>
+          )}
+        </Collection>
+      </TableBody>
+    </Table>
+  );
+};
+
+export const RenderEmptyStateStory  = {
+  render: RenderEmptyState,
+  args: {
+    isLoading: false
+  },
+  name: 'Empty/Loading Table rendered with TableLoader collection element'
+};
+
+interface Character {
+  name: string,
+  height: number,
+  mass: number,
+  birth_year: number
+}
+
+const OnLoadMoreTable = () => {
```

Comment:
Curiously, I just noticed that React complains about "React has detected a change in the order of Hooks called by Component" when the items are loaded... Will have to dig to see why that is